scripts/zones/Upper_Jeuno/npcs/Inconspicuous_Door.lua

```diff
@@ -0,0 +1,88 @@
+-----------------------------------
+-- Area: Upper Jeuno (244)
+--  NPC: Inconspicuous Door
+--  A Moogle Kupo d'Etat Mission NPC
+-----------------------------------
+require("scripts/globals/missions")
+-----------------------------------
+
+function onTrade(player,npc,trade)
+end
+
+function onTrigger(player,npc)
+
+    local jamInJeuno = player:getCurrentMission(AMK) == dsp.mission.id.amk.HASTEN_IN_A_JAM_IN_JEUNO
+    local myDecrepitDomicile = player:getCurrentMission(AMK) == dsp.mission.id.amk.WELCOME_TO_MY_DECREPIT_DOMICILE
+    local theProfessorsPrice = player:getCurrentMission(AMK) == dsp.mission.id.amk.AN_ERRAND_THE_PROFESSORS_PRICE
+    local abuseOfAuthority = player:getCurrentMission(AMK) == dsp.mission.id.amk.SHOCK_ARRANT_ABUSE_OF_AUTHORITY
+    local readTheFinePrint = player:getCurrentMission(AMK) == dsp.mission.id.amk.LENDER_BEWARE_READ_THE_FINE_PRINT
+    local aTriumphantReturn = player:getCurrentMission(AMK) == dsp.mission.id.amk.RELIEF_A_TRIUMPHANT_RETURN
+
+    local hasMetalStrip = player:hasKeyItem(dsp.ki.STURDY_METAL_STRIP)
+    local hasTreeBark = player:hasKeyItem(dsp.ki.PIECE_OF_RUGGED_TREE_BARK)
+    local hasLambRoast = player:hasKeyItem(dsp.ki.SAVORY_LAMB_ROAST)
+    local hasWormeatenChest = player:hasKeyItem(dsp.ki.MOLDY_WORMEATEN_CHEST)
+
+    if jamInJeuno then
+        player:startEvent(10178)
+        player:setVar("AMK4", 1)
+    elseif myDecrepitDomicile and hasMetalStrip and player:getVar("AMK4") == 1 then
+        player:startEvent(10179) -- Metal Strip handed in
+    elseif myDecrepitDomicile and player:getVar("AMK4") == 1 then
```

Comment:
Does the player have to obtain/give the KI in order?

Reply:
Once `myDecrepitDomicile` is flagged, the items can all be collected in any order, but they are handed in in order, as there are the specific reminders about what mog wants to be handed in next (cs: 10186, 10187 and 10188)